paper-linear-progress.html

```diff
@@ -0,0 +1,897 @@
+          animation-timing-function: cubic-bezier(.152313, .196432, .648374, 1.004315);
+          -webkit-transform: scaleX(.457104);
+          transform: scaleX(.457104);
+        }
+
+        44.15% {
+          -webkit-animation-timing-function: cubic-bezier(.257759, -0.003163, .211762, 1.381790);
+          animation-timing-function: cubic-bezier(.257759, -0.003163, .211762, 1.381790);
+          -webkit-transform: scaleX(.727960);
+          transform: scaleX(.727960);
+        }
+
+        100% {
+          -webkit-transform: scaleX(.08);
+          transform: scaleX(.08);
+        }
+      }
+
+      /**
+       * IE-specific translation animations. The normal rect translations do not
+       * work properly in IE 10/11 (they do however work properly in Edge) due to a
+       * bug involving resetting an elements position after each animation iteration
+       * (see: https://goo.gl/pquAMA). These are fallback animations that allow us to
+       * gracefully degrade in these browsers.
+       */
+
+      @-webkit-keyframes indeterminate-translate-ie {
+        from {
+          -webkit-transform: translateX(-100%);
+          transform: translateX(-100%);
+        }
+
+        to {
+          -webkit-transform: translateX(100%);
+          transform: translateX(100%);
+        }
+      }
+
+      @keyframes indeterminate-translate-ie {
+        from {
+          -webkit-transform: translateX(-100%);
+          transform: translateX(-100%);
+        }
+
+        to {
+          -webkit-transform: translateX(100%);
+          transform: translateX(100%);
+        }
+      }
+
+      @-webkit-keyframes indeterminate-translate-reverse-ie {
+        from {
+          -webkit-transform: translateX(100%);
+          transform: translateX(100%);
+        }
+
+        to {
+          -webkit-transform: translateX(-100%);
+          transform: translateX(-100%);
+        }
+      }
+
+      @keyframes indeterminate-translate-reverse-ie {
+        from {
+          -webkit-transform: translateX(100%);
+          transform: translateX(100%);
+        }
+
+        to {
+          -webkit-transform: translateX(-100%);
+          transform: translateX(-100%);
+        }
+      }
+    </style>
+
+    <div id="bufferingDots"></div>
+    <div id="buffer"></div>
+    <div class="progressBar" id="progressPrimary">
+      <span class="progressBarInner"></span>
+    </div>
+    <div class="progressBar" id="progressSecondary">
+      <span class="progressBarInner"></span>
+    </div>
+  </template>
+
+  <script>
+    Polymer({
+      is: 'paper-linear-progress',
+
+      behaviors: [
+        // NOTE(traviskaufman): This is needed because we have to use a var() for the animation.
+        // Until https://github.com/Polymer/polymer/issues/2399 is fixed, this cannot be done using
+        // var() within a keyframe block.
+        Polymer.NeonAnimationRunnerBehavior
```

Comment:
Ah never mind, I see you need it for `element.play/cancelAnimation()`, and you need that to customize `bgColor` of the SVG.

Not too familiar with SVG, but could you use variables on the CSS `fill` property instead (https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Fills_and_Strokes#Using_CSS)? That way instead of playing the animation with JS, you can just apply the animation by toggling a class or something.